nav2_dynamic_params test flaky.

## Bug report

**Required Info:**

- Operating System:
  - Ubuntu 18.04
- Installation type:
  - source
- Version or commit hash:
  - master latest as of 5 Feb 2019
- DDS implementation:
  - <!-- rmw_implementation used (e.g. Fast-RTPS, RTI Connext, etc -->
- Client library (if applicable):
  - <!-- e.g. rclcpp, rclpy, or N/A -->

#### Steps to reproduce issue
Run colcon test


#### Expected behavior
Tests pass

#### Actual behavior
Occasionally `nav2_dynamic_params` will fail. It has now failed in CircleCI as well as on my local machine.

Ouput looks like
```
build/nav2_dynamic_params/test_results/nav2_dynamic_params/test_dynamic_params_client.xml: 1 test, 0 errors, 1 failure, 0 skipped
- nav2_dynamic_params test_dynamic_params_client.missing_result
  <<< failure message
    The test did not generate a result file:
    
    [INFO] [launch]: process[test_dynamic_params_helper-1]: started with pid [17615]
    [INFO] [launch]: process[test_dynamic_params_client-2]: started with pid [17616]
    [ERROR] [launch]: process[test_dynamic_params_client-2] process has died [pid 17616, exit code 1, cmd '/opt/nav2_ws/build/nav2_dynamic_params/test/test_dynamic_params_client_exec'].
    [INFO] [launch]: sending signal 'SIGINT' to process[test_dynamic_params_helper-1]
    [INFO] [launch]: process[test_dynamic_params_helper-1]: process has finished cleanly
  >>>
```

#### Additional information

See https://circleci.com/gh/ros-planning/navigation2/16 for an example of a failing run.

--- a/tools/run_test_suite.bash
+++ b/tools/run_test_suite.bash
@@ -4,9 +4,26 @@ set -ex
 
 SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"  # gets the directory of this script
 
-colcon test --packages-skip nav2_system_tests
+# Skip nav2_system_tests because the system tests are flaky and need to be retried
+# a few times.
+#
+# Skip the nav2_dynamic_params tests because they fail when run concurrently with
+# other tests. We could run all the tests sequentially to fix this, however, by
+# just deferring this one package and running it later, we don't have to slow everything down.
+colcon test --packages-skip nav2_system_tests nav2_dynamic_params
+
+# run nav2_dynamic_params independently
+colcon test --packages-select nav2_dynamic_params
+
+# run the linters in nav2_system_tests. They only need to be run once.
 colcon test --packages-select nav2_system_tests --ctest-args --exclude-regex "test_.*"  # run the linters
+
+# Each of the `colcon test` lines above runs tests on independent sets of packages.
+# As a result the test logs of each line won't overwrite the others. The single
+# call to `colcon test-result` will look through all packages and report any errors
+# that happened in any of the `colcon test` lines above.
 colcon test-result --verbose
+
 $SCRIPT_DIR/ctest_retry.bash -r 3 -d build/nav2_system_tests -t test_localization
 $SCRIPT_DIR/ctest_retry.bash -r 3 -d build/nav2_system_tests -t test_simple_navigator
 $SCRIPT_DIR/ctest_retry.bash -r 3 -d build/nav2_system_tests -t test_bt_navigator

